test/bin/eshost.js

```diff
@@ -0,0 +1,379 @@
+'use strict';
+
+const assert = require('assert');
+const cp = require('child_process');
+const fs = require('fs');
+const path = require('path');
+const tokenize = require('yargs/lib/tokenize-arg-string');
+
+const Config = require('../../lib/config');
+
+const isWindows = process.platform === 'win32' ||
+  process.env.OSTYPE === 'cygwin' ||
+  process.env.OSTYPE === 'msys';
+
+const hosts = [
+  ['jsshell', { hostPath: 'js' }],
+  ['ch', { hostPath: 'ch' }],
+  ['node', { hostPath: 'node' }],
+  ['d8', { hostPath: 'd8' }],
+  ['jsc', { hostPath: 'jsc' }],
+];
+
+if (isWindows) {
+  hosts.forEach(record => {
+    record[1].hostPath += '.exe';
+  });
+}
+
+
+function eshost(command = []) {
+  let args = [];
+
+  if (Array.isArray(command)) {
+    args.push(...command);
+  } else {
+    if (typeof command === 'string') {
+      args.push(...tokenize(command));
+    }
+  }
+
+  return new Promise((resolve, reject) => {
+    let cps = cp.spawn('./bin/eshost.js', args);
+    let stdout = '';
+    let stderr = '';
+
+    cps.stdout.on('data', buffer => { stdout += buffer; });
+    cps.stderr.on('data', buffer => { stderr += buffer; });
+    cps.on('close', () => {
+      cps = null;
+      resolve({ stderr, stdout });
+    });
+
+    cps.on('error', error => {
+      reject(error);
+    });
+  });
+}
+
+let originalHostConfig;
+let nohosts = '{"hosts":{}}';
+try {
+  if (fs.existsSync(Config.defaultConfigPath)) {
+    originalHostConfig = fs.readFileSync(Config.defaultConfigPath, 'utf-8');
+  } else {
+    originalHostConfig = nohosts;
+  }
+} catch (error) {
+  originalHostConfig = nohosts;
+}
+
+function restoreOriginalHostConfig() {
+  fs.writeFileSync(Config.defaultConfigPath, originalHostConfig);
+}
+
+function emptyHostConfig() {
+  fs.writeFileSync(Config.defaultConfigPath, '{"hosts":{}}');
+}
+
+function toLines(result) {
+  return (result || '').split('\n').map(value => value.trim()).filter(Boolean);
+}
+
+function toHostPath(hostpath) {
+  return path.normalize(hostpath) + (isWindows ? '.exe' : '');
+}
+
+before(function() {
+  restoreOriginalHostConfig();
+});
+
+describe('eshost --help', () => {
+  it('displays help contents', () => {
+    return eshost('--help').then(result => {
+      assert.equal(result.stderr, '');
+
+      // TODO: Determine important things to assert about the help text
+      assert.notEqual(result.stdout, '');
+    });
+  });
+});
+
+describe('eshost --list', () => {
+
+  beforeEach(function() {
+    emptyHostConfig();
+  });
+
+  after(function() {
+    restoreOriginalHostConfig();
+  });
+
+  it('displays empty list when there are no hosts configured', () => {
+    return eshost('--list').then(result => {
+      assert.deepEqual(toLines(result.stdout), [
+        `Using config  ${Config.defaultConfigPath}`,
+        '┌──────┬──────┬──────┬──────┬──────┐',
+        '│ name │ type │ path │ args │ tags │',
```

Comment:
I would like _some_ way to test that 0 configured hosts produces _something_. Perhaps it would be better if *no table* is printed for that case, instead displaying something informative and I can test for that output.